lib/charms/kratos/v0/kratos_info.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for sharing kratos info.
+
+This library provides a Python API for both requesting and providing kratos deployment info,
+such as endpoints, namespace and ConfigMap details.
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+cd some-charm
+charmcraft fetch-lib charms.kratos.v0.kratos_info
+```
+To use the library from the requirer side:
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+requires:
+  kratos-info:
+    interface: kratos_info
+    limit: 1
+```
+Then, to initialise the library:
+```python
+from charms.kratos.v0.kratos_info import (
+    KratosInfoRelationError,
+    KratosInfoRequirer,
+)
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.kratos_info_relation = KratosInfoRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            kratos_data = self.kratos_info_relation.get_kratos_info()
+        except KratosInfoRelationError as error:
+            ...
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationCreatedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+
+# The unique Charmhub library identifier, never change it
+LIBID = "40d36890fe6d40409ccee34aa9245d4a"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 2
```

Comment:
```suggestion
LIBPATCH = 1
```

Reply:
This is from https://github.com/canonical/kratos-operator/pull/240/files#diff-63b726bbd10988b107e3a3f9ac68e469a6b64fa964a2ea9f70e734238ebc1943

Reply:
ahhh my bad